Source/WebCore/editing/ReplaceSelectionCommand.cpp

```diff
@@ -1384,7 +1385,8 @@ void ReplaceSelectionCommand::doApply()
 
     if (insertedNodes.isEmpty())
         return;
-    removeUnrenderedTextNodesAtEnds(insertedNodes);
+    if (isRichlyEditablePosition(insertionPos))
+        removeUnrenderedTextNodesAtEnds(insertedNodes);
```

Comment:
Is it worth have a comment explaining why we only do this for richly-editable text? e.g. something like:

`// removeUnrenderedTextNodesAtEnds can trigger layout, so for performance, only do it when the presence of richly-editable text requires us to.`

Reply:
@twilco - Added comment - no harm in it and might help in future as well.